llvm/lib/Target/EraVM/EraVMAsmPrinter.cpp

```diff
@@ -194,11 +205,29 @@ void EraVMAsmPrinter::emitEndOfAsmFile(Module &) {
     emitGlobalConstant(getDataLayout(), C);
   }
 
+  // Emit a section with the deafault landing pads.
+  emitDefaultLandingPads();
+
   // after emitting all the things, we also need to clear symbol cache
   UniqueConstants.clear();
   ConstantPoolMap.clear();
 }
 
+void EraVMAsmPrinter::insertSymbolToConstantMap(const Constant *C,
+                                                MCSymbol *Sym) {
+  auto result = std::find(UniqueConstants.begin(), UniqueConstants.end(), C);
```

Comment:
It seems this function could be done substantially shorter as
* `result` is used once so could become a temporary
* same for `symbol_vector`
* early return is preferable to if/else.

Reply:
Thank you, will update it.